Src/Eiffel/eiffel/byte_code/visitor/assertion_breakable_slot_strategy.e

```diff
@@ -0,0 +1,481 @@
+			process_elsif_expression_b,
+			process_if_b,
+			process_if_expression_b,
+			process_inspect_b,
+			process_instr_call_b,
+			process_loop_b,
+			process_loop_expr_b,
+			process_nested_b,
+			process_retry_b,
+			process_reverse_b,
+			process_check_b,
+			process_invariant_b,
+			process_guard_b
+		end
+
+	SHARED_BYTE_CONTEXT
+
+feature -- Status report
+
+	breakable_slot_count (a_node: BYTE_NODE; ctx: BYTE_CONTEXT): INTEGER
+			-- Does `a_node' contain an assignment to an attribute?
+		do
+			counter := 0
+			a_node.process (Current)
+			Result := counter
+
+			print (ctx.class_type.associated_class.name)
+			print (".")
+			print (ctx.current_feature.e_feature.name_8)
+			print (" -> " + a_node.generator + ": ")
+			print (Result)
+			print ("%N")
+
+		end
+
+feature {NONE} -- Implementation: access
+
+	counter: INTEGER
+
+	increment_counter (a_node: BYTE_NODE)
+		do
+			counter := counter + 1
+			if attached {ASSERT_B} a_node as a_assert and then attached a_assert.tag as t then
+				print (" - " + a_node.generator + " %"" + t + "%" [" + counter.out + "]%N")
+			else
+				print (" - " + a_node.generator + " [" + counter.out + "]%N")
+			end
+		end
+
+	generate_melted_debugger_hook_nested
+		do
+
+		end
+
+feature -- Node processing
+
+	process_assert_b (a_node: ASSERT_B)
+			-- Process `a_node`.
+			-- (from BYTE_NODE_ITERATOR)
+			-- (export status {BYTE_NODE})
+		do
+				-- Assertion mark
+			if context.assertion_type = {ASSERT_TYPE}.in_precondition then
+				increment_counter (a_node)
+				a_node.expr.process (Current)
+			else
+				if context.assertion_type /= {ASSERT_TYPE}.in_invariant then
+						-- No hooks when it is an invariant
+					increment_counter (a_node)
+				end
+					-- Assertion byte code
+				a_node.expr.process (Current)
+			end
+		end
+
+	process_inv_assert_b (a_node: INV_ASSERT_B)
+			-- Process `a_node'.
+		do
+			process_assert_b (a_node)
+		end
+
+	process_require_b (a_node: REQUIRE_B)
+		do
+			process_assert_b (a_node)
+		end
+
+	process_variant_b (a_node: VARIANT_B)
+			-- Process `a_node'.
+		do
+			process_assert_b (a_node)
+		end
+
+	process_check_b (a_node: CHECK_B)
+			-- Process `a_node'.
+		do
+			if a_node.check_list /= Void then
+					-- Set assertion type
+				context.set_assertion_type ({ASSERT_TYPE}.In_check)
+				a_node.check_list.process (Current)
+				context.set_assertion_type (0)
+			end
+		end
+
+
+	process_guard_b (a_node: GUARD_B)
+			-- <Precursor>
+		do
+				-- Generate hook for the condition test.
+			if attached a_node.check_list as c then
+					-- Generated byte code for assertions.
+				context.set_assertion_type ({ASSERT_TYPE}.In_guard)
+				c.process (Current)
+				context.set_assertion_type (0)
+			end
+			if attached a_node.compound as c then
+					-- Generated byte code for compound.
+				c.process (Current)
+			end
+		end
+
+	process_invariant_b (a_node: INVARIANT_B)
+			-- Process `a_node'.
+		local
+			l_context: like context
+		do
+			l_context := context
```

Comment:
Given that `context` is a once, saving it to a local might not be very efficient, because it adds additional stack manipulations.